tedge_config/src/tedge_config_location.rs

```diff
@@ -0,0 +1,80 @@
+use std::path::PathBuf;
+
+const DEFAULT_ETC_PATH: &str = "/etc";
+const TEDGE_CONFIG_FILE: &str = "tedge.toml";
+
+/// Information about where `tedge.toml` is located and the defaults that are based
+/// on that location.
+///
+/// Broadly speaking, we distinguish two different locations:
+///
+/// - System-wide locations under `/etc/tedge`
+/// - User-local locations under `$HOME/.tedge`
+///
+/// We DO NOT base the defaults on the currently executing user. Instead, we base
+/// the defaults on the location of the `tedge.toml` file. If it is located in
+/// `/etc`, regardless of the executing user, we use defaults that use system-wide
+/// locations (e.g. `/etc/ssl/certs`). Whereas if `tedge.toml` is located in a users
+/// home directory, we base the defaults on locations within the users home directory.
+///
+/// This allows run `sudo tedge -c '$HOME/.tedge/tedge.toml ...` where the defaults are picked up
+/// correctly.
+///
+/// The choice, where to find `tedge.toml` on the other hand is based on the executing user AND the
+/// env `$HOME`.  But once we have found `tedge.toml`, we never again have to care about the
+/// executing user (except when `chown`ing files...).
+///
+#[derive(Debug, Clone)]
+pub struct TEdgeConfigLocation {
+    /// Full path to `tedge.toml`.
+    pub tedge_config_path: PathBuf,
+
+    /// Default device cert path
+    pub default_device_cert_path: PathBuf,
+
+    /// Default device key path
+    pub default_device_key_path: PathBuf,
+
+    /// Default path for azure root certificates
+    pub default_azure_root_cert_path: PathBuf,
+
+    /// Default path for c8y root certificates
+    pub default_c8y_root_cert_path: PathBuf,
```

Comment:
I practice we can have a single default here for well-known root certificates in general.